### ISSUE TITLE
date-picker: handle RangeError

### ISSUE THREAD

**Issue by OrbisK** [TuSimple/naive-ui#3296]
### TuSimple/naive-ui version (版本)
2.31.0

### Vue version (Vue 版本)
3.2.37

### Browser and its version (浏览器及其版本)
OperaGX (88.0.4412.85)

### System and its version (系统及其版本)
Win 11

### Reappearance link (重现链接)
https://codesandbox.io/s/quizzical-pond-uf6873?file=/src/Demo.vue

### Reappearance steps (重现步骤)
Open link to minimal reproduction

### Expected results (期望的结果)
The app should not crash completely. It might show an error or leave the datepicker empty?

### Actual results (实际的结果)
The app is crashing.

**Comment by Sepush** [comment]
**foo** is not a correct time string that't why the app will crash
![image](https://user-images.githubusercontent.com/39197136/179245598-7c61f412-fd7c-4f90-9cdc-332f4a5b08d1.png)

**Comment by OrbisK** [comment]
> **foo** is not a correct time string that't why the app will crash ![image](https://user-images.githubusercontent.com/39197136/179245598-7c61f412-fd7c-4f90-9cdc-332f4a5b08d1.png)

Yes, sure, but I think in some scenarios there is a possibility that a timestamp is not formatted correctly. If so, naive ui is freezing completely.

**Comment by 07akioni** [comment]
> > **foo** is not a correct time string that't why the app will crash ![image](https://user-images.githubusercontent.com/39197136/179245598-7c61f412-fd7c-4f90-9cdc-332f4a5b08d1.png)
> 
> Yes, sure, but I think in some scenarios there is a possibility that a timestamp is not formatted correctly. If so, naive ui is freezing completely. Wouldn't it be better if there was some kind of fallback?

Make it break will let user know bugs in their code at early stage.

Imagine you pass a wrong type data in date picker and there's no any error or warning, it would lead to more bug that hard to find later.

**Comment by OrbisK** [comment]
> > > **foo** is not a correct time string that't why the app will crash ![image](https://user-images.githubusercontent.com/39197136/179245598-7c61f412-fd7c-4f90-9cdc-332f4a5b08d1.png)
> > 
> > 
> > Yes, sure, but I think in some scenarios there is a possibility that a timestamp is not formatted correctly. If so, naive ui is freezing completely. Wouldn't it be better if there was some kind of fallback?
> 
> Make it break will let user know bugs in their code at early stage.
> 
> Imagine you pass a wrong type data in date picker and there's no any error or warning, it would lead to more bug that hard to find later.

I fully agree with this point.

However, I have a usecase where I want to load data dynamically from an Excel. Theoretically x columns should be formatted according to a format (timestamp). Should the user swap a column, or enter the wrong format, then the application would simply no longer be usable for the user.
